toolchain/check/import_ref.cpp

```diff
@@ -19,6 +19,33 @@ namespace Carbon::Check {
 
 // Resolves an instruction from an imported IR into a constant referring to the
 // current IR.
+//
+// Calling Resolve on an instruction operates in an iterative manner, tracking
+// Work items on work_stack_. At a high level, the loop is:
+//
+// 1. If Work has received a constant, it's considered resolved.
+//    - If incomplete_type is marked, resolve unconditionally.
+//    - The constant check avoids performance costs of deduplication on add.
+// 2. Resolve the instruction: (TryResolveInst/TryResolveTypedInst)
+//    A. For cases with incomplete types, when incomplete_type isn't marked:
+//      i. Make the incomplete type with associated constant.
+//        - If the imported type is incomplete, return the constant.
+//      ii. Mark incomplete_type.
+//    B. Gather all input constants.
+//      - Gathering constants directly adds unresolved values to work_stack_.
+//    C. If any need to be resolved (HasNewWork), return Invalid; this
+//       instruction needs two calls to complete.
+//    D. Build any necessary IR structures, and return the output constant.
+//    - For trivial cases with zero or one input constants, this may return
+//      a constant (if one, potentially Invalid) directly.
```

Comment:
I think this is missing an indent?
```suggestion
//      - For trivial cases with zero or one input constants, this may return
//        a constant (if one, potentially Invalid) directly.
```

Reply:
Restructuring with "For most cases:" to make the intended overlap clear.